src/main/java/org/takes/http/BkTimeable.java

```diff
@@ -0,0 +1,171 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.http;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Back decorator with maximum lifetime.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.14.2
+ */
+@EqualsAndHashCode(of = {"origin", "latency" })
+public final class BkTimeable implements Back {
+
+    /**
+     * Original back.
+     */
+    private final transient Back origin;
+
+    /**
+     * Maximum latency in milliseconds.
+     */
+    private final transient long latency;
+
+    /**
+     * Threads storage.
+     */
+    private final transient Set<BkTimeable.ThreadInfo> threads;
+
+    /**
+     * Monitoring executor.
+     */
+    private final transient ScheduledExecutorService service;
+
+    /**
+     * Ctor.
+     * @param back Original back
+     * @param msec Execution latency
+     */
+    public BkTimeable(final Back back, final long msec) {
+        this(
+            back,
+            msec,
+            Executors.newSingleThreadScheduledExecutor(),
+            new CopyOnWriteArraySet<BkTimeable.ThreadInfo>()
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param back Original back
+     * @param msec Execution latency
+     * @param svc Executor service
+     * @param map Threads storage
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    BkTimeable(final Back back, final long msec,
+        final ScheduledExecutorService svc,
+        final Set<BkTimeable.ThreadInfo> map) {
+        this.origin = back;
+        this.latency = msec;
+        this.threads = map;
+        this.service = svc;
+        this.start();
+    }
+
+    @Override
+    public void accept(final Socket socket) throws IOException {
+        this.threads.add(new BkTimeable.ThreadInfo());
+        this.origin.accept(socket);
+    }
+
+    /**
+     * Start monitoring.
+     * @checkstyle MagicNumberCheck (25 lines)
```

Comment:
I would place this right before `100`. now you're suppressing 25 lines, it's not healthy